10-kubernetes-inputs.tf

```
@@ -96,3 +96,48 @@ variable "azure_policy_enabled" {
   description = "Enable the Azure Policy addon"
   default     = false
 }
+
+variable "node_os_maintenance_window_config" {
+  type = object({
+    frequency   = string
+    interval    = number
+    duration    = number
+    day_of_week = optional(string)
+    start_time  = optional(string)
+    utc_offset  = optional(string)
+    start_date  = optional(string)
+  })
+  default = {
+    frequency   = "Weekly"
+    interval    = 1
+    duration    = 4
+    day_of_week = "Monday"
+    start_time  = "18:00"
+    utc_offset  = "+00:00"
+    start_date  = null
+  }
+
+  validation {
+    condition     = var.node_os_maintenance_window_config.duration >= 4
+    error_message = "Maintenance window duration must be at least 4 hours when node_os_channel_upgrade is enabled."
+  }
+
+  validation {
+    condition     = try(contains(["Daily", "Weekly"], var.node_os_maintenance_window_config.frequency), false)
+    error_message = "Maintenance window frequency must be set to 'Daily' or 'Weekly'."
+  }
+
+  validation {
+    condition     = var.node_os_maintenance_window_config.interval >= 1
+    error_message = "Maintenance window interval must be at least 1."
+  }
+
+  # If Daily is set it must not have day of week. If Weekly is set it must have a day of week set.
+  validation {
+    condition = (
+      var.node_os_maintenance_window_config.frequency == "Daily" && var.node_os_maintenance_window_config.day_of_week == null) || (
+      var.node_os_maintenance_window_config.frequency == "Weekly" &&
+    try(contains(["Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday"], var.node_os_maintenance_window_config.day_of_week), false))
+    error_message = var.node_os_maintenance_window_config.frequency == "Daily" ? "'day_of_week' must not be set for 'Daily' frequency." : "Invalid 'day_of_week', please choose a day of the week."
```

```suggestion
      try(contains(["Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday"], var.node_os_maintenance_window_config.day_of_week), false))
      error_message = var.node_os_maintenance_window_config.frequency == "Daily" ? "'day_of_week' must not be set for 'Daily' frequency." : "Invalid 'day_of_week', please choose a day of the week."
```

All inline would be nice if it's not going to error